Annotato/Annotato/Views/Document/Edit/DocumentView.swift

```diff
@@ -123,7 +212,35 @@ extension DocumentView {
         return visiblePages.contains(pageContainingAnnotation)
     }
 
+    private func selectionBoxIsInVisiblePages(selectionBox: SelectionBoxView, visiblePages: [PDFPage]) -> Bool {
+        guard let centerInDocument = pdfView.documentView?.convert(selectionBox.center, to: self) else {
+            return false
+        }
+        guard let pageContainingSelectionBox = pdfView.page(for: centerInDocument, nearest: true) else {
+            return false
+        }
+        return visiblePages.contains(pageContainingSelectionBox)
+    }
+
+    private func linkLineIsInVisiblePages(linkLine: LinkLineView, visiblePages: [PDFPage]) -> Bool {
+        guard let centerInDocument = pdfView.documentView?.convert(linkLine.center, to: self) else {
+            return false
+        }
+        guard let pageContainingLinkLine = pdfView.page(for: centerInDocument, nearest: true) else {
+            return false
+        }
+        return visiblePages.contains(pageContainingLinkLine)
+    }
+
```

Comment:
Is it possible to extract these 3 functions into 1? Seems like they're doing the same thing. You can pass in different views to the same function.

```suggestion
    private func isInVisiblePages(view: UIView, visiblePages: [PDFPage]) -> Bool {
        guard let centerInDocument = pdfView.documentView?.convert(view.center, to: self) else {
            return false
        }
        guard let pageContainingView = pdfView.page(for: centerInDocument, nearest: true) else {
            return false
        }
        return visiblePages.contains(pageContainingView)
    }
```
Also, could this be part of an extension of `PDFView`? If yes, then we can use the one below

```
extension PDFView {
    private func visiblePagesContains(view: UIView) -> Bool {     
        guard let centerInDocument = self.documentView?.convert(view.center, to: self) else {
            return false
        }
        guard let pageContainingView = self.page(for: centerInDocument, nearest: true) else {
            return false
        }
        return self.visiblePages.contains(pageContainingView)
    }
}
```

Reply:
Ok done